copy/copy.go

```diff
@@ -463,6 +463,7 @@ func (c *copier) copyMultipleImages(ctx context.Context, policyContext *signatur
 		if err != nil {
 			return nil, "", errors.Wrapf(err, "Error converting manifest list to list with MIME type %q", selectedListType)
 		}
+		listIsModified = true
```

Comment:
This is easy to forget in any future updates, and we are already using `reflect.DeepEqual` above. Could the `if listIsModified` be replaced with something like `reflect.DeepEqual(list, original)`? That would solve the problem once for all.

Using `reflect` is ugly in that it will also compare private fields, but at least it fails safe (preferring a conversion even if not necessary); we could also add a `List.Equals()` method, but that would have the same risk of forgetting to compare some members.

Reply:
Would it be simpler to serialize both, and then compare the byte slices?

Reply:
Good idea, that would work nicely. (The Go implementation does not _promise_ that, but the JSON marshaller serializes fields in the order they are declared, and anyway, it would fail safe as well.)

Reply:
Trying that.